dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,239 @@
+    const { controller, name } = publicData;
+    const creationDate = publicData.metadata["dataset.created"];
+    return [controller, name, creationDate];
+  };
+  let sortedRepositories = publicData;
+  if (activeSortIndex !== null) {
+    sortedRepositories = publicData.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (aValue === bValue) {
+        return 0;
+      }
+      if (activeSortDirection === "asc") {
+        return aValue > bValue ? 1 : -1;
+      } else {
+        return bValue > aValue ? 1 : -1;
+      }
+    });
+  }
+  const getSortParams = (columnIndex) => ({
+    isFavorites: columnIndex === 3,
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex,
+  });
+  const handleButtonClick = (_isSelected, event) => {
+    const id = event.currentTarget.id;
+    setIsSelected(id);
+  };
+  const setControllerName = (controllerNameValue) => {
+    controllerName = controllerNameValue;
+  };
+  const setDateRange = (startDateValue, endDateValue) => {
+    startDate = startDateValue;
+    endDate = endDateValue;
+  };
+  const saveFavorites = (fav) => {
+    localStorage.setItem("favControllers", JSON.stringify(fav));
+  };
+  const Sidebar = (
+    <PageSidebar
+      nav={<NavItems />}
+      className="sidebar"
+      isNavOpen={isNavOpen}
+    />
+  );
+  const NavbarDrawer = () => {
+    return (
+      <Masthead id="basic">
+        <MastheadToggle>
+          <PageToggleButton isNavOpen={isNavOpen} onNavToggle={onNavToggle}>
+            <BarsIcon />
+          </PageToggleButton>
+        </MastheadToggle>
+      </Masthead>
+    );
+  };
+  const selectedArray =
+    isSelected === "firstButton" ? publicData : favoriteRepoNames;
+  return (
+    <>
+      <Page header={<NavbarDrawer />} sidebar={Sidebar}>
+        <AlertMessage
+          message="Want to see only metric relevant to you?"
+          link="Login to create an account"
+        />
+        <PageSection variant={PageSectionVariants.light}>
+          <PathBreadCrumb pathList={["Dashboard", "Components"]} />
+          <Heading headingTitle="Controllers"></Heading>
+          <div className="filterContainer">
+            <SearchBox
+              dataArray={dataArray}
+              setPublicData={setPublicData}
+              startDate={startDate}
+              endDate={endDate}
+              setControllerName={setControllerName}
+            />
+            <DatePickerWidget
+              dataArray={dataArray}
+              setPublicData={setPublicData}
+              controllerName={controllerName}
+              setDateRange={setDateRange}
+            />
+          </div>
+          <ToggleGroup aria-label="Available options with Single Selectable">
```

Comment:
So aria-label attribute is designed to help assistive technologies (for example,lets say a screen-reader) attach a label to otherwise anonymous HTML Elements.

So ,suppose there is a <label> element

<label for="username">Type your username</label>
<input type="text">

Here label clearly indicates that a user has to type in a username.

But lets say we have a html element where it is not practical to have a label on screen,then aria-label attribute comes in handy.

For example ,we have a lets say a button to close the dialog box.

<button>X</button>

Now ,this is fine for a person who has no physical disabilities.But a blind person who has a screen reader on,will only hear "X",and he might not be able to make out that this is a close button.

So, instead we can put an aria-label in this case

<button aria-label="Close dialog box">X</button>

This is the main purpose of aria-label.

Reply:
Thanks for the excellent explanation!

So, should the `aria-label` be something more like "Results selection options..."  ("Available options" seems like it lacks important context.)